SingularityService/src/main/java/com/hubspot/singularity/data/StateManager.java

```diff
@@ -296,9 +297,18 @@ public SingularityState generateState(boolean includeRequestIds) {
 
     int numDeploys = 0;
     long oldestDeploy = 0;
+    long oldestDeployStep = 0;
+    List<SingularityDeployMarker> activeDeploys = new ArrayList<>();
     final long now = System.currentTimeMillis();
 
     for (SingularityPendingDeploy pendingDeploy : deployManager.getPendingDeploys()) {
+      activeDeploys.add(pendingDeploy.getDeployMarker());
+      if (pendingDeploy.getDeployProgress().isPresent() && !pendingDeploy.getDeployProgress().get().isStepComplete()) {
+        long deployStepDelta = now - pendingDeploy.getDeployProgress().get().getTimestamp();
+        if (deployStepDelta > oldestDeployStep) {
+          oldestDeployStep = deployStepDelta;
+        }
+      }
       long delta = now - pendingDeploy.getDeployMarker().getTimestamp();
       if (delta > oldestDeploy) {
         oldestDeploy = delta;
```

Comment:
two different deltas. I'm adding additional fields since it is likely some things still use the old `oldestDeploy` one, so still calculating that one as normal